tests/utils/test_io.py

```diff
@@ -0,0 +1,260 @@
+import os
+import io
+import pytest
+from prompt_toolkit.document import Document
+from prompt_toolkit.validation import ValidationError
+
+import rasa.utils.io as io_utils
+
+os.environ["USER_NAME"] = "user"
+os.environ["PASS"] = "pass"
+
+
+def test_read_yaml_string():
+    config_without_env_var = """
+    user: user
+    password: pass
+    """
+    r = io_utils.read_yaml(config_without_env_var)
+    assert r["user"] == "user" and r["password"] == "pass"
+
+
+def test_read_yaml_string_with_env_var():
+    config_with_env_var = """
+    user: ${USER_NAME}
+    password: ${PASS}
+    """
+    r = io_utils.read_yaml(config_with_env_var)
+    assert r["user"] == "user" and r["password"] == "pass"
+
+
+def test_read_yaml_string_with_multiple_env_vars_per_line():
+    config_with_env_var = """
+    user: ${USER_NAME} ${PASS}
+    password: ${PASS}
+    """
+    r = io_utils.read_yaml(config_with_env_var)
```

Comment:
@arthurTemporim had some comments about the naming of `r` . If you have time you could give that a proper name, e.g. `parsed` or `result`